### PR TITLE
feat: apply op-geth v1.101105.1

### PR DESCRIPTION
**Pull request by 0xHansLee**
### Description

Unlike the previous work as [@chokobole done](https://github.com/kroma-network/go-ethereum/commit/b5ba1126882230084ca94f1d05c7b98af0bda191), I applied op-geth [v1.101105.1](https://github.com/ethereum-optimism/op-geth/releases/tag/v1.101105.1) onto `dev` branch of kroma geth.

First of all, thank you for your understanding that I made commits one by one. 
Since I'm not familiar with geth, it was helpful to me to follow all the changes commit by commit. This PR will be squashed and merged into `dev` branch after approved.

I also do not grasp all the changes in detail, so if you have any question or find any weired points, please leave comments.

Please review this PR carefully with the following main changes:
- `CurrentBlock()` in `core/blockchain_reader.go` returns `*types.Header`, not `*types.Block` anymore (and related funcs as well)
- refactoring of `core.Message` in `core/state_transition.go`
- add field of `EffectiveGasPrice` to `Receipt` in `core/types/receipt.go`

From what I understand, there is no critical changes related to ZK trie.

### Ref commits

- [core/rawdb: find smallest block stored in key-value store when chain](https://github.com/ethereum-optimism/op-geth/commit/f7661a662a6e0743b440956e4e15bdae1ba3821e)
- [signer/core: accept all solidity primitive types for EIP-712 signing](https://github.com/ethereum-optimism/op-geth/commit/02796f6bee7e014fd16ad39f0bcd3b665b51e0bb)
- [params: remove EF azure bootnodes](https://github.com/ethereum-optimism/op-geth/commit/e14043db71c5d2d91520fab217302fcecf7aa939)
- [core/vm: use golang native big.Int](https://github.com/ethereum-optimism/op-geth/commit/b80f05bde2c4e93ae64bb3813b6d67266b5fc0e6)
- [core/vm: fix typo in comment](https://github.com/ethereum-optimism/op-geth/commit/1bf1168432223b69de97975bb457870889a85dd3)
- [core/forkid: fix issue in validation test](https://github.com/ethereum-optimism/op-geth/commit/5395362e0f2ca6f3c9a3a2aa7efd14bfae5a86da)
- [cmd/evm: update readmes for the tests](https://github.com/ethereum-optimism/op-geth/commit/08f6a2a89d89e4510ab98ad8fc4e6b1894d0da94)
- [core, core/types: plain Message struct](https://github.com/ethereum-optimism/op-geth/commit/67ac5f0ae797fbafa1e57e982cab4c2d291244c3)
- [travis: only build PPAs nightly, not on every push, too heavy](https://github.com/ethereum-optimism/op-geth/commit/df027995437d84ae8f0476a1be42a3dba64edae1)
- [p2p: small comment typo](https://github.com/ethereum-optimism/op-geth/commit/051493d9bf39dd55e152915082cc2eeeccd23096)
- [core: add Timestamp method in BlockGen](https://github.com/ethereum-optimism/op-geth/commit/564db9a95f564f06566072d429761f7f2d4ee2da)
- [core/txpool: implement additional DoS defenses](https://github.com/ethereum-optimism/op-geth/commit/6cf2e921a79c2ac914f8b413f3c689919a60110f)
- [tests: define MuirGlacier fork (#26856](https://github.com/ethereum-optimism/op-geth/commit/789de23d16045190af3ace2276ea3957190e869b)
- [code/vm: fix comment typo](https://github.com/ethereum-optimism/op-geth/commit/ca61048178bd25bdb42adb026c1e1639864bcffd)
- [core: minor code refactor](https://github.com/ethereum-optimism/op-geth/commit/a20e38720c239a8e62797ec0ff8ffbcb813b2300)
- [core/txpool: use priceList.Put instead of heap.Push](https://github.com/ethereum-optimism/op-geth/commit/d1c5f918a3d2e332d63725566a2ac54a616fa7e8)
- [eth: return error if 'safe' or 'finalized' tag used pre-merge](https://github.com/ethereum-optimism/op-geth/commit/5f81db68c6cd0544ae5f3b9b4c7b1d56d02b23da)
- [.travis.yml: reenable PPA build on tag push](https://github.com/ethereum-optimism/op-geth/commit/94ff7219114540263c76d1895618041959fae1f9)
- [core/state, trie: port changes from PBSS](https://github.com/ethereum-optimism/op-geth/commit/c8a6b7100c56255a9cde580be59136beb8d28b8e)
- [p2p/discover: pass invalid discv5 packets to Unhandled channel](https://github.com/ethereum-optimism/op-geth/commit/eca3d39c314c0eaa2b2b616619fde84d032a11b1)
- [all: update links in documentation](https://github.com/ethereum-optimism/op-geth/commit/b5c9be3358f0eb0a6f217474678dfd609f1d2743)
- [Increase websocket frame size (from erigon rpc client)](https://github.com/ethereum-optimism/op-geth/commit/6bc68f8d94b359e08beefa2b6006e5ad003519f9)
- [Add missing L1CostFunc set](https://github.com/ethereum-optimism/op-geth/commit/e95b13f657be5a2722a85d2457598f8e072b365a)
- [cmd/devp2p, cmd/geth: add version in --help output](https://github.com/ethereum-optimism/op-geth/commit/f86913bc3e9a4f2439b6c3cd4d00cb364495238c)
- [core: show db error-info in case of mismatched hash root](https://github.com/ethereum-optimism/op-geth/commit/bba2a1bac5fb896e644ff6a5307419812e0f6819)
- [consensus: improve consensus engine definition](https://github.com/ethereum-optimism/op-geth/commit/48d1bf06780799304b96d8c0fea3cc61941da4df)
- [eth/catalyst: increase update consensus timeout](https://github.com/ethereum-optimism/op-geth/commit/d8066dcde801baed8ebc1605bd3bfaefa781e8e3)
- [internal/ethapi: avoid int overflow in GetTransactionReceipt](https://github.com/ethereum-optimism/op-geth/commit/f73365738309d9b3249efca9c1492856bf6e7848)
- [core: Set L1CostFunc in NewEVMBlockContext](https://github.com/ethereum-optimism/op-geth/commit/e3c5831e7cdb722c83bbb12142e673749f79ddd3)
- [Merge tag 'v1.11.4' into update-geth-vana](https://github.com/ethereum-optimism/op-geth/commit/70234f9b682ac7fdc7f12fca6ec611c340ec9940)
- [trie, accounts/abi: add error-checks](https://github.com/ethereum-optimism/op-geth/commit/b7bfbc1e649cecc9fa11ad55b3943272a919c24b)
- [rlp: support for uint256](https://github.com/ethereum-optimism/op-geth/commit/58d0f6440ba82a0cbf327ca77b7cd795c2d687c6)
- [eth: fix output file permissions in admin_exportChain](https://github.com/ethereum-optimism/op-geth/commit/ee8e83fa5f6cb261dad2ed0a7bbcde4930c41e6c)
- [trie: reduce unit test time](https://github.com/ethereum-optimism/op-geth/commit/81b0aa0cc705e5583ba3f03ff7f3dba5965f7d49)
- [core/txpool: use atomic int added in go1.19](https://github.com/ethereum-optimism/op-geth/commit/80ff0b4e6a1b3059195ad7949a2e5dbf12065592)
- [params: schedule shanghai fork on mainnet](https://github.com/ethereum-optimism/op-geth/commit/5d23d21fffa8d11e599d8e482323c321a88658f4)
- [params: go-ethereum v1.11.5 stable](https://github.com/ethereum-optimism/op-geth/commit/a38f4108571d1a144dc3cf3faf8990430d109bc4)
- [Merge tag 'v1.11.5' into aj/update-geth-erszbat](https://github.com/ethereum-optimism/op-geth/commit/22354a944a891959bfceaa1122c98765b95b1c6a)
- [optimism: finalize PoS on startup to enable full Merge functionality](https://github.com/ethereum-optimism/op-geth/commit/555b76f39878ba72c6faf5c7ad6219f3222762ff)
- [miner: Account for L1 Info Deposit gas usage in the pending pool](https://github.com/ethereum-optimism/op-geth/commit/f856aea527d9dd87cfee6f922b4a6749cb774942)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -62,3 +62,4 @@ issues:
     - 'SA1019: strings.Title is deprecated'
     - 'SA1019: strings.Title has been deprecated since Go 1.18 and an alternative has been available since Go 1.0: The rule Title uses for word boundaries does not handle Unicode punctuation properly. Use golang.org/x/text/cases instead.'
     - 'SA1029: should not use built-in type string as key for value'
+    - 'SA1019: mrand.Seed has been deprecated since Go 1.20 and an alternative has been available since Go 1.0'
```
Comment (0xHansLee) [inline .golangci.yml:65]: There was a lint error related `mrand.See` which is used in `zk_trie_proof_test.go`, so I excluded.
Please check @ccrayz 🙏

**Comment by 0xHansLee** [comment]
Two testdata, `suicide.json` and `selfdestruct.json` under `eth/tracers/internal/tracetest/testdata/call_tracer_flat`, are deleted because we do not support `SELFDESTRUCT` opcode for now.

File: core/state_transition.go [hunk 2]
```diff
@@ -355,87 +378,78 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
...
 
-	// if deposit: skip refunds, skip tipping coinbase
-	if st.msg.IsDepositTx() {
-		// Record deposits as using all their gas (matches the gas pool)
-		return &ExecutionResult{
-			UsedGas:    st.msg.Gas(),
-			Err:        vmerr,
-			ReturnData: ret,
-		}, nil
-	}
```
Comment (0xHansLee) [inline core/state_transition.go:414]: This code block seems to be duplicated with the below, so I deleted it.

Comment (chokobole) [reply core/state_transition.go:414]: As mentioned in the comment, this code block intentionally doesn't tip to the `coinbase` (a.k.a `proposerFeeRecipient`) and also skips refunding to the `from` address in the case of deposit transactions, so it should be kept as is.

File: core/types/rollup_l1_cost.go [hunk 3]
```diff
@@ -33,18 +33,13 @@ func (r RollupGasData) DataGas() (gas uint64) {
...
-}
-
 type StateGetter interface {
 	GetState(common.Address, common.Hash) common.Hash
 }
 
 // L1CostFunc is used in the state transition to determine the cost of a rollup message.
 // Returns nil if there is no cost.
-type L1CostFunc func(blockNum uint64, msg RollupMessage) *big.Int
+type L1CostFunc func(blockNum uint64, blockTime uint64, dataGas RollupGasData, isDepositTx bool) *big.Int
```
Comment (chokobole) [inline core/types/rollup_l1_cost.go:42]: It seems like `blockTime` argument is unnecessary. Could you remove it?

**Comment by 0xHansLee** [comment]
The author of all commits in this PR is all me. In order to bring the intact information of the commits, I will close this PR and open a new PR.

**Pull request closed without merging**